feat: new caido extension methodcheck

GangGreenTemperTatum opened the pull request:
# I am submitting a new Plugin Package 🍰 

## Repository URL

https://github.com/GangGreenTemperTatum/MethodCheck

## Release Checklist

- [x] I have tested the plugin on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] My GitHub release contains all required files
  - [x] `plugin_package.zip`
  - [x] `plugin_package.zip.sig`
- [x] GitHub Tag name matches the exact version number specified in my `manifest.json`
- [x] The `id` in my `manifest.json` matches the `id` in the `plugin_packages.json` file.
- [x] My `README.md` describes the plugin package purpose and provides clear usage instructions.
- [x] I have read the developer policy at <https://developer.caido.io/policy.html>, and have assessed my plugin package adherence to this policy.
- [x] I have added a license in the `LICENSE` file and it matches the `license` field in the `plugin_packages.json` file.
- [x] My project respects and is compatible with the original license of any third-party code that I'm using.
      I have given proper attribution to these other projects in my `README.md` and/or `LICENSE`.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/caido/store?pullRequest=24) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/caido/store?pullRequest=24) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/caido/store?pullRequest=24) it.</sub>

Sytten commented:
Thanks for the submission, I will review it when I am back home on Tuesday :)

GangGreenTemperTatum commented:
> Thanks for the submission, I will review it when I am back home on Tuesday :) 

Thank you! :) safe travels!!

GangGreenTemperTatum commented:
hey @Sytten ! thanks so much for your awesome support and review yesterday - i did a stream of [issues/PRs](https://github.com/GangGreenTemperTatum/MethodCheck/commits/main/) yesterday and this morning and believe i've hopefully addresses your concerns (also included some typechecks in a github workflow etc too :))

i ran a new [release workflow this morning](https://github.com/GangGreenTemperTatum/MethodCheck/actions/runs/14944689198/job/41986498105) and re-published `v1.0.0` - any chance i could ask you to take another look when you have a second pls? tyia and have a great weekend!! 😁

Sytten commented:
Author withdrew submission for now since the code needs more work
